microsoft/featuremanagement/_featuremanager.py

```diff
@@ -41,6 +36,20 @@ def __init__(self, feature_flags, **kwargs):
                 raise ValueError("Custom filter must be a subclass of FeatureFilter")
             self._filters[filter.name] = filter
 
+    def _get_feature_flag(self, feature_flag_name):
+        feature_management = self._configuration.get(FEATURE_MANAGEMENT_KEY)
+        if not feature_management:
+            return None
+        feature_flags = feature_management.get(FEATURE_FLAG_KEY)
+        if not feature_flags:
+            return None
+
+        for feature_flag in feature_flags:
+            if feature_flag.get("id") == feature_flag_name:
+                return FeatureFlag.convert_from_json(feature_flag)
```

Comment:
It looks to me like every time someone calls `is_enabled` there is a conversion from JSON, is that right?

Reply:
If so, let's track an item to cache that.

Reply:
Yes, that because we need to support the provider refreshing, but can't guarantee that the customer gave us json as a string vs a json/Map object.

Reply:
We can do better. There should be some happy path out there where is_enabled is very cheap.

Reply:
How? We just have a pointer to the "configuration" object. How would we know if it has changed or not? I guess we could hash the whole configuration and check if it changes?

Otherwise we have no real way to tell. We could just use the json and not the object. We use to do that, but was changed to a PR comment.

Reply:
Javascript will likely have the same concern. Whatever we land on should be applied to both.

Reply:
> Javascript will likely have the same concern. Whatever we land on should be applied to both.

So far in JavaScript FM design, JSON parsing happens in provider.
Calling `is_enabled` only searches array of flags by name, which is not expensive as I tested before.

> Performance of array.find() : linear search, the worst case is ~19ms per 1 million flags. For feature flag number < 10K (should be large enough for most cases), the time cost < 1ms. 
Tested on CPU: Intel(R) Core(TM) i7-10700 CPU @ 2.90GHz